### ISSUE TITLE
[BUG] Some files have not passed the integrity check. Upgrading v27 -> v28.0.0

### ISSUE THREAD

**Issue by nemesis1337** [linuxserver/docker-nextcloud#386]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

After upgrading to v28.0.0

In Nextcloud Administrator settings -> Security & setup warnings
-> Some files have not passed the integrity check. 
 
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- INVALID_HASH
		- core/js/mimetypelist.js

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [core/js/mimetypelist.js] => Array
                        (
                            [expected] => 550ab566d30693bfa24ec4b15d9df87731ae8a3be8f79dabf94757e5b8b20eec6e4b678f17af1718297f2872f6b04519eeb024d1dff11947f29da431c7f11201
                            [current] => 24e6faa4a1361e51ff70027024f41c4b37fdb5f6f347ceb095450ddbde2d0a6b6dd1403be92a912d2889064a9736a9bed89d7dc62470035fc1299d4b977c5f46
                        )

                )

        )

)

### Expected Behavior

No error or all files passing the integrity check

### Steps To Reproduce

1. go to Administrative settings
2. overview 
3. show in "Security & setup warnings" when the scan finishes running

### Environment

```markdown
- OS: Ubuntu 22.04.3 LTS (GNU/Linux 6.5.11-6-pve x86_64)
- How docker service was installed: using portainer with a stack
```

### CPU architecture

x86-64

### Docker creation

```bash
Branch 	Version   Built
latest 	28.0.0 	  12 Dec 2023 at 13:04:11 

portainer stack:
version: "2.1"
services:
  nextcloud:
    image: linuxserver/nextcloud:latest
    container_name: nextcloud
    environment:
      - TZ=Europe/Amsterdam
    volumes:
      - nc_config:/config
      - nc_data:/data
    networks:
      - nextcloud_default
      - local-net
    ports:
      - 444:443
    restart: unless-stopped
```

### Container logs

```bash
Updating <dav> ...
Fix broken values of calendar objects
 Starting ...
Clean up old calendar subscriptions from deleted users that were not cleaned-up
    0/0 [>---------------------------]   0% Starting ...
Updated <dav> to 1.29.1
Updating <encryption> ...
Updated <encryption> to 2.16.0
Updating <files_sharing> ...
Updated <files_sharing> to 1.20.0
Updating <files_trashbin> ...
Updated <files_trashbin> to 1.18.0
Updating <files_versions> ...
Updated <files_versions> to 1.21.0
Updating <sharebymail> ...
Updated <sharebymail> to 1.18.0
Updating <workflowengine> ...
Updated <workflowengine> to 2.10.0
Updating <comments> ...
Updated <comments> to 1.18.0
Updating <firstrunwizard> ...
Updated <firstrunwizard> to 2.17.0
Updating <nextcloud_announcements> ...
Updated <nextcloud_announcements> to 1.17.0
Updating <notifications> ...
Updated <notifications> to 2.16.0
Updating <systemtags> ...
Updated <systemtags> to 1.18.0
Updating <theming> ...
Migrate old user accessibility config
    0/0 [->--------------------------]   0% Starting ...
Updated <theming> to 2.3.0
Updating <bruteforcesettings> ...
Updated <bruteforcesettings> to 2.8.0
Updating <contactsinteraction> ...
Updated <contactsinteraction> to 1.9.0
Updating <dashboard> ...
Updated <dashboard> to 7.8.0
Updating <federatedfilesharing> ...
Updated <federatedfilesharing> to 1.18.0
Updating <files_pdfviewer> ...
Updated <files_pdfviewer> to 2.9.0
Updating <files_reminders> ...
Updated <files_reminders> to 1.1.0
Updating <logreader> ...
Updated <logreader> to 2.13.0
Updating <privacy> ...
Updated <privacy> to 1.12.0
Updating <provisioning_api> ...
Updated <provisioning_api> to 1.18.0
Updating <related_resources> ...
Updated <related_resources> to 1.3.0
Updating <serverinfo> ...
Updated <serverinfo> to 1.18.0
Updating <settings> ...
Updated <settings> to 1.10.0
Updating <support> ...
Updated <support> to 1.11.0
Updating <survey_client> ...
Updated <survey_client> to 1.16.0
Updating <text> ...
Updated <text> to 3.9.1
Updating <twofactor_backupcodes> ...
Updated <twofactor_backupcodes> to 1.17.0
Updating <twofactor_totp> ...
Updated <twofactor_totp> to 10.0.0-beta.2
Updating <user_status> ...
Updated <user_status> to 1.8.1
Updating <viewer> ...
Updated <viewer> to 2.2.0
Updating <weather_status> ...
Updated <weather_status> to 1.8.0
Update app contacts from App Store
Update app integration_giphy from App Store
Update app passwords from App Store
Update app spreed from App Store
Starting code integrity check...
Finished code integrity check
Update successful
Turned off maintenance mode
Resetting log level
Initializing finished
**** The following active confs have different version dates than the samples that are shipped. ****
**** This may be due to user customization or an update to the samples. ****
**** You should compare the following files to the samples in the same folder and update them. ****
**** Use the link at the top of the file to view the changelog. ****
┌────────────┬────────────┬────────────────────────────────────────────────────────────────────────┐
│  old date  │  new date  │ path                                                                   │
├────────────┼────────────┼────────────────────────────────────────────────────────────────────────┤
│ 2021-10-24 │ 2023-04-13 │ /config/nginx/nginx.conf                                               │
│ 2022-08-20 │ 2023-08-13 │ /config/nginx/ssl.conf                                                 │
│            │ 2023-06-23 │ /config/nginx/site-confs/default.conf                                  │
└────────────┴────────────┴────────────────────────────────────────────────────────────────────────┘
[custom-init] No custom files found, skipping...
[ls.io-init] done.
PDOException: SQLSTATE[HY000]: General error: 2006 MySQL server has gone away in /app/www/public/3rdparty/doctrine/dbal/src/Driver/PDO/Statement.php:130
      
Stack trace:
#0 /app/www/public/3rdparty/doctrine/dbal/src/Driver/PDO/Statement.php(130): PDOStatement->execute()
#1 /app/www/public/3rdparty/doctrine/dbal/src/Connection.php(1202): Doctrine\DBAL\Driver\PDO\Statement->execute()
#2 /app/www/public/lib/private/DB/Connection.php(294): Doctrine\DBAL\Connection->executeStatement()
#3 /app/www/public/3rdparty/doctrine/dbal/src/Query/QueryBuilder.php(386): OC\DB\Connection->executeStatement()
#4 /app/www/public/lib/private/DB/QueryBuilder/QueryBuilder.php(280): Doctrine\DBAL\Query\QueryBuilder->execute()
#5 /app/www/public/lib/private/DB/QueryBuilder/QueryBuilder.php(326): OC\DB\QueryBuilder\QueryBuilder->execute()
#6 /app/www/public/lib/private/BackgroundJob/JobList.php(367): OC\DB\QueryBuilder\QueryBuilder->executeStatement()
#7 /app/www/public/lib/private/BackgroundJob/JobList.php(355): OC\BackgroundJob\JobList->unlockJob()
#8 /app/www/public/cron.php(158): OC\BackgroundJob\JobList->setLastJob()
      
}
Next Doctrine\DBAL\Driver\PDO\Exception: SQLSTATE[HY000]: General error: 2006 MySQL server has gone away in /app/www/public/3rdparty/doctrine/dbal/src/Driver/PDO/Exception.php:28
      
Stack trace:
#0 /app/www/public/3rdparty/doctrine/dbal/src/Driver/PDO/Statement.php(132): Doctrine\DBAL\Driver\PDO\Exception::new()
#1 /app/www/public/3rdparty/doctrine/dbal/src/Connection.php(1202): Doctrine\DBAL\Driver\PDO\Statement->execute()
#2 /app/www/public/lib/private/DB/Connection.php(294): Doctrine\DBAL\Connection->executeStatement()
#3 /app/www/public/3rdparty/doctrine/dbal/src/Query/QueryBuilder.php(386): OC\DB\Connection->executeStatement()
#4 /app/www/public/lib/private/DB/QueryBuilder/QueryBuilder.php(280): Doctrine\DBAL\Query\QueryBuilder->execute()
#5 /app/www/public/lib/private/DB/QueryBuilder/QueryBuilder.php(326): OC\DB\QueryBuilder\QueryBuilder->execute()
#6 /app/www/public/lib/private/BackgroundJob/JobList.php(367): OC\DB\QueryBuilder\QueryBuilder->executeStatement()
#7 /app/www/public/lib/private/BackgroundJob/JobList.php(355): OC\BackgroundJob\JobList->unlockJob()
#8 /app/www/public/cron.php(158): OC\BackgroundJob\JobList->setLastJob()
      
}
Next Doctrine\DBAL\Exception\ConnectionLost: An exception occurred while executing a query: SQLSTATE[HY000]: General error: 2006 MySQL server has gone away in /app/www/public/3rdparty/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:104
      
Stack trace:
#0 /app/www/public/3rdparty/doctrine/dbal/src/Connection.php(1938): Doctrine\DBAL\Driver\API\MySQL\ExceptionConverter->convert()
#1 /app/www/public/3rdparty/doctrine/dbal/src/Connection.php(1880): Doctrine\DBAL\Connection->handleDriverException()
#2 /app/www/public/3rdparty/doctrine/dbal/src/Connection.php(1208): Doctrine\DBAL\Connection->convertExceptionDuringQuery()
#3 /app/www/public/lib/private/DB/Connection.php(294): Doctrine\DBAL\Connection->executeStatement()
#4 /app/www/public/3rdparty/doctrine/dbal/src/Query/QueryBuilder.php(386): OC\DB\Connection->executeStatement()
#5 /app/www/public/lib/private/DB/QueryBuilder/QueryBuilder.php(280): Doctrine\DBAL\Query\QueryBuilder->execute()
#6 /app/www/public/lib/private/DB/QueryBuilder/QueryBuilder.php(326): OC\DB\QueryBuilder\QueryBuilder->execute()
#7 /app/www/public/lib/private/BackgroundJob/JobList.php(367): OC\DB\QueryBuilder\QueryBuilder->executeStatement()
#8 /app/www/public/lib/private/BackgroundJob/JobList.php(355): OC\BackgroundJob\JobList->unlockJob()
#9 /app/www/public/cron.php(158): OC\BackgroundJob\JobList->setLastJob()
#10 {main}
Next OC\DB\Exceptions\DbalException: An exception occurred while executing a query: SQLSTATE[HY000]: General error: 2006 MySQL server has gone away in /app/www/public/lib/private/DB/Exceptions/DbalException.php:71
      
Stack trace:
#0 /app/www/public/lib/private/DB/QueryBuilder/QueryBuilder.php(328): OC\DB\Exceptions\DbalException::wrap()
#1 /app/www/public/lib/private/BackgroundJob/JobList.php(367): OC\DB\QueryBuilder\QueryBuilder->executeStatement()
#2 /app/www/public/lib/private/BackgroundJob/JobList.php(355): OC\BackgroundJob\JobList->unlockJob()
#3 /app/www/public/cron.php(158): OC\BackgroundJob\JobList->setLastJob()
      
}
PHP Fatal error:  Uncaught RedisException: Redis server redis:6379 went away in /app/www/public/lib/private/Memcache/Redis.php:193
      
Stack trace:
#0 /app/www/public/lib/private/Memcache/Redis.php(193): Redis->evalsha()
#1 /app/www/public/lib/private/Memcache/Redis.php(177): OC\Memcache\Redis->evalLua()
#2 /app/www/public/lib/private/Lock/MemcacheLockingProvider.php(82): OC\Memcache\Redis->cad()
#3 /app/www/public/lib/private/Lock/AbstractLockingProvider.php(101): OC\Lock\MemcacheLockingProvider->releaseLock()
#4 [internal function]: OC\Lock\AbstractLockingProvider->releaseAll()
      
}
  thrown in /app/www/public/lib/private/Memcache/Redis.php on line 193
    0/0 [--->------------------------]   0%      [migrations] started
[migrations] 01-nginx-site-confs-default: skipped
[migrations] 02-default-location: skipped
[migrations] done
usermod: no changes
───────────────────────────────────────
      ██╗     ███████╗██╗ ██████╗ 
      ██║     ██╔════╝██║██╔═══██╗
      ██║     ███████╗██║██║   ██║
      ██║     ╚════██║██║██║   ██║
      ███████╗███████║██║╚██████╔╝
      ╚══════╝╚══════╝╚═╝ ╚═════╝ 
   Brought to you by linuxserver.io
───────────────────────────────────────
To support LSIO projects visit:
https://www.linuxserver.io/donate/
───────────────────────────────────────
GID/UID
───────────────────────────────────────
User UID:    911
User GID:    911
───────────────────────────────────────
using keys found in /config/keys
**** The following active confs have different version dates than the samples that are shipped. ****
**** This may be due to user customization or an update to the samples. ****
**** You should compare the following files to the samples in the same folder and update them. ****
**** Use the link at the top of the file to view the changelog. ****
┌────────────┬────────────┬────────────────────────────────────────────────────────────────────────┐
│  old date  │  new date  │ path                                                                   │
├────────────┼────────────┼────────────────────────────────────────────────────────────────────────┤
│ 2021-10-24 │ 2023-04-13 │ /config/nginx/nginx.conf                                               │
│ 2022-08-20 │ 2023-08-13 │ /config/nginx/ssl.conf                                                 │
│            │ 2023-06-23 │ /config/nginx/site-confs/default.conf                                  │
└────────────┴────────────┴────────────────────────────────────────────────────────────────────────┘
[custom-init] No custom files found, skipping...
[ls.io-init] done.
```

**Comment by cheesynator** [comment]
I can confirm this error. 
I also get an error regarding the hash value in the mimetypelist.js file.
Furthermore, I get error messages during the check which are constantly increasing. However, nothing is displayed in the log section.
I have gone back to version 27, where there are no error messages.

**Comment by Stooovie** [comment]
Same issue with the same file

**Comment by Maginos2210** [comment]
Same issue with same file. I run the docker on unraid.

**Comment by Maginos2210** [comment]
Possible Fix?

In [this](https://www.truenas.com/community/threads/nextcloud-integrity-check-failed-after-update.101041/post-699876) topic I found the link to the official [downloads of nextcloud](https://download.nextcloud.com/server/releases/).

Then I downloaded version 28.0.0 and extracted it.

Then I copied the file nextcloud-28.0.0/nextcloud/core/js/mimetypelist.js to my unraid server 

On my unraid server, I shut down the nextcloud container and went into the nextcloud folder: 

cd /mnt/user/appdata/nextcloud/nextcloud/core/js

mv mimetypelist.js mimetypelist.js.bak

Then I copied the mimetypelist.js file I copied earlier on the server into the /mnt/user/appdata/nextcloud/nextcloud/core/js folder and started the nextcloud container. After a rescan in the administration settings the error was gone. 

I don't know, if this is the best solution, but at least the error is gone.
